backend/dataall/modules/redshift_datasets_shares/aws/redshift_data.py

```diff
@@ -0,0 +1,408 @@
+        self.database = connection.database
+        self.execute_connection_params = {
+            'Database': connection.database,
+        }
+        if connection.workgroup:
+            self.execute_connection_params['WorkgroupName'] = connection.workgroup
+        if connection.clusterId:
+            self.execute_connection_params['ClusterIdentifier'] = connection.clusterId
+        if connection.secretArn:
+            self.execute_connection_params['SecretArn'] = connection.secretArn
+        if connection.redshiftUser and connection.clusterId:
+            # https://boto3.amazonaws.com/v1/documentation/api/1.26.93/reference/services/redshift-data/client/list_databases.html
+            # We cannot use DbUser with serverless for role federation.
+            # It must use the current session IAM role, which in this case would be the pivot role.
+            self.execute_connection_params['DbUser'] = connection.redshiftUser
+
+    def _execute_statement(self, sql: str):
+        log.info(f'Executing {sql=} with connection {self.execute_connection_params}...')
+        execute_dict = self.execute_connection_params
+        execute_dict['Sql'] = sql
+        execute_statement_response = self.client.execute_statement(**execute_dict)
+
+        execution_finished = False
+        describe_statement_response = None
+        while not execution_finished:
+            describe_statement_response = self.client.describe_statement(Id=execute_statement_response['Id'])
+            time.sleep(1)
+            execution_finished = describe_statement_response['Status'] not in ['PICKED', 'STARTED', 'SUBMITTED']
+
+        if describe_statement_response['Status'] == 'FAILED':
+            raise Exception(describe_statement_response['Error'])
+
+        log.info(f'Received response {describe_statement_response["Id"]}')
+        return describe_statement_response['Id']
+
+    def _execute_statement_return_records(self, sql: str):
+        id = self._execute_statement(sql=sql)
+        log.info(f'Returning records for sql {id=}...')
+        try:
+            response = self.client.get_statement_result(Id=id)
+            next_token = response.get('NextToken', None)
+            records = response.get('Records', [])
+            while next_token:
+                response = self.client.get_statement_result(Id=response['Id'], NextToken=next_token)
+                new_records = response.get('Records', [])
+                records.extend(new_records)
+                next_token = response.get('NextToken', None)
+            filtered_records = [[d for d in record if d.get('stringValue', False)] for record in records]
+            log.info(f'Returning {len(filtered_records)} records from executed statement')
+            return filtered_records
+        except Exception as e:
+            log.error(f'Failed to retrieve records for sql {id=}: {e}')
+            raise e
+
+    @staticmethod
+    def parsed_name(name: str) -> str:
+        return f'"{name}"'
```

Comment:
nit: should this be named something other than parsed like `quoted_name` ? Also should we have an equivalent function for names that must be single quoted as well?

Reply:
Done! We don't but maybe we should, there are a number of places that require single quotes. I'll add that